src/e3/anod/qualifiers_manager.py

```diff
@@ -1,40 +1,210 @@
 from __future__ import annotations
 
-from typing import TYPE_CHECKING, cast
+from typing import TYPE_CHECKING
 from hashlib import sha1
-
 from e3.anod.error import AnodError
+import abc
+import re
+import logging
 
-import string
 
 if TYPE_CHECKING:
-    from typing import (
-        Literal,
-        TypedDict,
-    )
     from e3.anod.spec import Anod
 
-    # Declare types
-    TAG_QUALIFIER_TYPE = Literal["tag"]
-    KEY_VALUE_QUALIFIER_TYPE = Literal["key-value"]
 
-    class QualifierDeclaration(TypedDict):
-        # Fields common to all qualifiers
-        description: str
-        test_only: bool
-        repr_in_hash: bool
-        repr_alias: str
+VALID_NAME = re.compile(r"^[a-zA-Z0-9_.-]+$")
+
+
+def check_valid_name(name: str, value_kind: str, origin: str) -> str:
+    """Check if a value is a valid qualifier name or component name.
+
+    A valid name is any non empty string containg alphanumeric character, dot,
+    dashes and underscores.
+
+    :param name: the name to check
+    :param error_msg: an error msg
```

Comment:
Hmmmm, does not correspond to the actual parameters :)

Reply:
docstring updated